src/privatesend-client.cpp

```diff
@@ -1452,7 +1452,24 @@ bool CPrivateSendClientSession::CreateDenominated(CConnman& connman)
 
     bool fCreateMixingCollaterals = !pwalletMain->HasCollateralInputs();
 
-    for (const auto& item : vecTally) {
+    // NOTE: We do not allow txes larger than 100kB, so we have to limit number of inputs here.
+    // We still want to consume a lot of inputs to avoid creating only smaller denoms though.
+    // Knowing that each CTxIn is at least 148b big, 400 inputs should take 400 x ~148b = ~60kB.
+    // This still leaves more than enough room for another data of typical CreateDenominated tx.
+    const size_t MAX_INPUTS = 400;
```

Comment:
This seems bad, assuming that an input will be 148 bytes, what about when improvements come along and decrease that somehow, or increase it somehow...

Reply:
Hmmm.. I don't understand what you are trying to say tbh.

Reply:
Ah, no, this one is a pretty stable one - it is also used to calculate dust for example.